scripts/build_components.sh

```diff
@@ -24,56 +35,59 @@ if [ -z "${tags}" ]; then
   exit 1
 fi
 
-# Set github repo information
-namespace="ml6team"
-repo="fondant"
+# Set default values for optional arguments if not passed
+component="${component:-all}"
+components_dir="${components_dir:-components}"
+namespace="${namespace:-ml6team}"
+repo="${repo:-fondant}"
+
 
 # Get the component directory
 scripts_dir=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
 root_dir=$(dirname "$scripts_dir")
-component_dir=$root_dir/"components"
+components_dir=$root_dir/${components_dir}
 
 # Loop through all subdirectories
-for dir in "$component_dir"/*/; do
+for dir in "$components_dir"/*/; do
   pushd "$dir"
 
   BASENAME=${dir%/}
   BASENAME=${BASENAME##*/}
+  if [[ "$BASENAME" == "${component}" ]] || [[ "${component}" == "all" ]]; then
```

Comment:
Can we do this a bit different? Python pseudo code:

```
components = [dir in components_dir] if component == "all" else [component]
print(f"building components {components}")
for component in components:
    ...
```

I think this would be easier to understand and debug.